clinica/pipelines/pet_linear/pet_linear_utils.py

```diff
@@ -134,61 +134,129 @@ def crop_nifti(input_img: str, ref_img: str) -> str:
+
+def _get_bids_entities_without_suffix(filename: str, suffix: str) -> str:
+    """Return the BIDS entities without the suffix from a BIDS path."""
     import os
 
-    from nipype.interfaces.utility import Rename
     from nipype.utils.filemanip import split_filename
 
-    _, source_file_pet, _ = split_filename(in_bids_pet)
-
-    # Rename into CAPS PET:
-    rename_pet = Rename()
-    rename_pet.inputs.in_file = fname_pet
-    if not uncropped_image:
-        suffix = f"_space-MNI152NLin2009cSym_desc-Crop_res-1x1x1_suvr-{suvr_reference_region}_pet.nii.gz"
-        rename_pet.inputs.format_string = source_file_pet + suffix
-    else:
-        suffix = f"_space-MNI152NLin2009cSym_res-1x1x1_suvr-{suvr_reference_region}_pet.nii.gz"
-        rename_pet.inputs.format_string = source_file_pet + suffix
-    out_caps_pet = rename_pet.run().outputs.out_file
-
-    # Rename into CAPS transformation file:
-    rename_trans = Rename()
-    rename_trans.inputs.in_file = fname_trans
-    rename_trans.inputs.format_string = source_file_pet + "_space-T1w_rigid.mat"
-    out_caps_trans = rename_trans.run().outputs.out_file
-
-    # Rename intermediate PET in T1w MRI space
-    if fname_pet_in_t1w is not None:
-        rename_pet_in_t1w = Rename()
-        rename_pet_in_t1w.inputs.in_file = fname_pet_in_t1w
-        rename_pet_in_t1w.inputs.format_string = (
-            source_file_pet + "_space-T1w_pet.nii.gz"
-        )
-        out_caps_pet_in_t1w = rename_pet_in_t1w.run().outputs.out_file
-    else:
-        out_caps_pet_in_t1w = None
+    fpath, stem, _ = split_filename(filename)
+    return os.path.join(fpath, stem.rstrip(f"_{suffix}"))
+
+
+def _rename_pet_into_caps(
+    entities: str, filename: str, cropped: bool, suvr_reference_region: str
+) -> str:
+    """Rename into CAPS PET."""
+    return _rename(filename, entities, _get_pet_suffix(cropped, suvr_reference_region))
+
+
+def _rename_transformation_into_caps(entities: str, filename: str) -> str:
+    """Rename into CAPS transformation file."""
+    return _rename(filename, entities, "_space-T1w_rigid.mat")
+
+
+def _rename_intermediate_pet_in_t1w_space_into_caps(
+    entities: str, filename: str
+) -> str:
+    """Rename intermediate PET in T1w MRI space."""
+    return _rename(filename, entities, "_space-T1w_pet.nii.gz")
+
+
+def _rename(filename: str, entities: str, suffix: str):
+    """Rename 'filename' into '{entities}{suffix}'."""
+    from nipype.interfaces.utility import Rename
+
+    rename = Rename()
+    rename.inputs.in_file = filename
+    rename.inputs.format_string = entities + suffix
+
+    return rename.run().outputs.out_file
+
+
+def _get_pet_suffix(cropped: bool, suvr_reference_region: str) -> str:
+    space = "_space-MNI152NLin2009cSym"
+    resolution = "_res-1x1x1"
+    desc = "_desc-Crop" if cropped else ""
+    suvr = f"_suvr-{suvr_reference_region}"
 
-    return out_caps_pet, out_caps_trans, out_caps_pet_in_t1w
+    return f"{space}{desc}{resolution}{suvr}_pet.nii.gz"
```

Comment:
Perhaps `pet_bids_components` and specifying what we mean by components in the docstring.

That being said, it might be local enough to the code that it won't end up too disturbing. It's not like we are using this taxonomy wrongly everywhere.

Reply:
Sounds good ! Done in [5bbd017](https://github.com/aramis-lab/clinica/pull/935/commits/5bbd01746ca7ee0aeab0fed7ea191e967d000c32)